### ISSUE TITLE
Electron

### ISSUE THREAD

**Issue by TonyCois** [BrestJS/community#1]
Faire une conférence / rex sur Electron

(Via Discord)

**Comment by shimaore** [comment]
Ça je veux bien faire (et ça fait longtemps que j'ai pas présenté ☺ ).

**Comment by shimaore** [comment]
Thomas(? je crois) avait donné (en fin de soirée) des détails sur ce qu'il attendait, entre autres des informations sur les performances, l'architecture etc. Killian avait un début de réponse avec l'importance du noyau (qui fait le lien entre Node et Chromium), une comparaison avec XULRunner, la nécessité d'intégrer du C++ pour les endroits "critiques", le fait que beaucoup d'application "construites avec Electron" sont construites avec des builds d'Electron qui ne sont pas la mainline…

**Comment by Thomasorus** [comment]
Yes, pour moi la conf devrait être avant tout être orientée pratique et la technique associée. Après une introduction de l'histoire du framework et de pourquoi il existe, ce serait pas mal d'essayer de se mettre dans la peau d'un cas pratique. 

Mettons tu es une société qui fait du logiciel web et a qui on demande une version desktop de l'app pour des raisons diverses, est-ce que partir sur Electron serait un bon choix ? Quels sont les points techniques qui font que oui et non ? Qu'est-ce qui va bloquer ? Je vois que Slack et VSCode ils ont super rapides et opti, je peux faire pareil facilement avec ma grosse app pas forcément optimisée, ou il y a une embrouille quelque part ?

**Comment by othelarian** [comment]
J'ai vu mon nom, de la lumière, je suis entré ...
J'ai un gros REX sur electron, avec deux usages, sur des systèmes embarqués en entrepôt, avec de IP65, connecteurs en or et tout et tout. Mais !

En ce moment je planche pas mal sur GTK et son webkit intégré pour faire fonctionner Miso (Haskell powa) avec un backend Haskell derrière (on ne change pas une équipe qui gagne), et j'ai aussi dans les cartons ma conf du DevFest de Brest avec Qt, et son système QML/js, et son intégration de webkit.

Et j'ai aussi travaillé sur Xul, avec Xulrunner (ma tendre jeunesse, c'était il y a ...). C'était fun, mais pas terrible pour plein de raisons.

Je propose donc un gros REX sur electron / nw (que j'utilise toujours, mais pas en dév), Xulrunner (Firefox, Thunderbird), Qt et GTK ou pourquoi ça ne prend pas, sauf electron. Par contre c'est un REX, pas un code lunch, et montrer du code dans autant de technos différentes n'a pas de sens. Je peux parler de cas pratiques (avec un 's' pour le coup XD), mais revenir sur du code dans chaque branche est un investissement trop important en terme de temps. Donc si quelqu'un veut s'associer pour parler plutôt technique sur certains sujets, welcome ^^ .

**Comment by shimaore** [comment]
J'étais partant pour faire une présentation ou un atelier, mais sans doute pas à un niveau très approfondi (en particulier pas au niveau d'expérience que demande Thomas :) ) — ma seule expérience c'est monter une appli temps réel pour faire du mapping MIDI pour un Launchpad Mini pour mon fiston (ça a l'avantage d'être coloré, en tout cas).

(Quoique j'ai sans doute un début de projet de portage de mon UI de provisioning vers Electron qui doit traîner qq part. Si je le finis j'aurai peut-être plus de choses à dire.)

J'ai joué avec Qt il y a très longtemps (`moc` :duck: `moc`), et je vois en gros à quoi ressemble la boucle d'évènements de Gtk mais ça ne va pas plus loin. Mais peut-être que ça suffit?

Bon en tout cas je veux bien donner un coup de main.

**Comment by othelarian** [comment]
Eh, on reparle de ce projet ? J'ai d'autres billes dans mon sac depuis ;-) Je suis prêt pour une présentation / REX, juste besoin de savoir pour une date.

Sinon, @shimaore , on pourrait se compléter, moi un historique et un REX, toi un atelier sur electron, qu'en penses-tu ? Là je suis sur revery, un concurrent, mais vu qu'il faut d'abord des connaissances en Ocaml/ReasonML ça me paraît chaud d'attaquer directement dessus.

**Comment by shimaore** [comment]
(@othelarian et moi avions discuté sur discord, donc on est en sync.)

Pour la partie atelier-Electron je n'ai pas encore commencé mais je pense que je me concentrerai sur les spécificités d'Electron (par exemple, créer des menus, utiliser l'IPC pour le dialogue entre la partie UI et le backend Node.js, etc.) plutôt qu'essayer de faire une appli complète et vraiment utile.

Le truc dont il faudra se méfier c'est qu'une vingtaine de personnes en train de faire `npm install electron` en même temps ça risque de saturer rapidement le réseau du site où on fera l'atelier. ;)
